content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md

```diff
@@ -0,0 +1,214 @@
+typically using the type defined by a {{< glossary_tooltip
+term_id="CustomResourceDefinition" text="CRDs" >}} that was created when
+installing a resource driver.
+
+The `core/v1` `PodSpec` defines ResourceClaims that are needed for a Pod in a new
+`resourceClaims` field. Entries in that list reference either a ResourceClaim
+or a ResourceClaimTemplate. When referencing a ResourceClaim, all Pods using
+this PodSpec (for example, inside a Deployment or StatefulSet) share the same
+ResourceClaim instance. When referencing a ResourceClaimTemplate, each Pod gets
+its own instance.
+
+The `resources.claims` list for container resources define whether a container gets
+access to these resource instances, which makes it possible to share resources
+between one or more containers.
+
+Here is an example for a fictional resource driver. Two ResourceClaim objects
+will get created for this Pod and each container gets access to one of them.
+
+```yaml
+apiVersion: resource.k8s.io/v1alpha1
+kind: ResourceClass
+name: resource.example.com
+driverName: resource-driver.example.com
+---
+apiVersion: cats.resource-driver.example.com/v1
+kind: ClaimParameters
+name: resource-claim-parameters
+color: black
+size: large
+---
+apiVersion: resource.k8s.io/v1alpha1
+kind: ResourceClaimTemplate
+metadata:
+  name: resource-claim-template
+spec:
+  spec:
+    resourceClassName: resource.example.com
+    parametersRef:
+      apiGroup: cats.resource-driver.example.com
+      kind: ClaimParameters
+      name: resource-claim-parameters
+–--
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+spec:
+  containers:
+  - name: container0
+    image: ubuntu:20.04
+    command: ["sleep", "9999"]
+    resources:
+      claims:
+      - name: resource0
+  - name: container1
+    image: ubuntu:20.04
+    command: ["sleep", "9999"]
+    resources:
+      claims:
+      - name: resource1
+  resourceClaims:
+  - name: resource0
+    source:
+      resourceClaimTemplateName: resource-claim-template
+  - name: resource1
+    source:
+      resourceClaimTemplateName: resource-claim-template
+```
+
+## Scheduling
+
+In contrast to native resources (CPU, RAM) and extended resources (managed by a
+device plugin, advertised by kubelet), the scheduler has no knowledge of what
+dynamic resources are available in a cluster or how they could be split up to
+satisfy the requirements of a specific ResourceClaim. Resource drivers are
+responsible for that. They mark ResourceClaims as "allocated" once resources
+for it are reserved. This also then tells the scheduler where in the cluster a
+ResourceClaim is available.
+
+ResourceClaims can get allocated as soon as they are created ("immediate
+allocation"), without consider which Pods will use them. The default is to
+delay allocation until a Pod gets scheduled which needs the ResourceClaim
+("wait for first consumer").
+
+In that mode, the scheduler checks all ResourceClaims needed by a Pod and
```

Comment:
General comment -- does it make sense to add backticks around all of the object references e.g. `ResourceClaims`?

Reply:
No, that doesn't seem to get done elsewhere either, for example https://kubernetes.io/docs/concepts/storage/persistent-volumes/ uses PersistentVolumeClaim without any markup except when first introducing it.